Test/VariantSearchTests.cs

```diff
@@ -141,5 +142,55 @@ public void MoreTests(string filename, DecoyType decoyType = DecoyType.None)
+            string raw = Path.Combine(TestContext.CurrentContext.TestDirectory, @"TestVariantPep.mzML");            
+            EverythingRunnerEngine noVariants = new EverythingRunnerEngine(new List<(string, MetaMorpheusTask)> { ("NoVariantOutput", task) }, new List<string> { raw }, new List<DbForTask> { noVariantDb }, thisTaskOutputFolder);
+            EverythingRunnerEngine variants = new EverythingRunnerEngine(new List<(string, MetaMorpheusTask)> { ("VariantOutput", task) }, new List<string> { raw }, new List<DbForTask> { variantDb }, thisTaskOutputFolder);
+
+            noVariants.Run();
+            variants.Run();
+
+            // no variant results files should be generated
+            HashSet<string> expectedFiles = new HashSet<string> {
+                "AllPeptides.psmtsv", "AllPSMs.psmtsv", "AllPSMs_FormattedForPercolator.tsv", "prose.txt", "results.txt" };
+
+            HashSet<string> files1 = new HashSet<string>(Directory.GetFiles(Path.Combine(thisTaskOutputFolder, "NoVariantOutput")).Select(v => Path.GetFileName(v)));
+
+            // these 2 lines are for debug purposes, so you can see which files you're missing (if any)
+            var missingFiles = expectedFiles.Except(files1).ToList();
+            var extraFiles = files1.Except(expectedFiles).ToList();
+
+            // test that output is what's expected
+            //Assert.That(missingFiles.Count() == 0 && extraFiles.Count() ==0);
+
+            HashSet<string> files2 = new HashSet<string>(Directory.GetFiles(Path.Combine(thisTaskOutputFolder, "VariantOutput")).Select(v => Path.GetFileName(v)));
+            // variant files should be generates
+            expectedFiles = new HashSet<string> {
+                "AllPeptides.psmtsv", "AllPSMs.psmtsv", "AllPSMs_FormattedForPercolator.tsv", "AllProteinGroups.tsv", "AllQuantifiedPeaks.tsv", "AllQuantifiedPeptides.tsv", "prose.txt", "results.txt", "VariantPeptides.psmtsv", "VariantAnalysisResultSummary.txt", "VariantPSMs.psmtsv" };
+
+            // these 2 lines are for debug purposes, so you can see which files you're missing (if any)
+            missingFiles = expectedFiles.Except(files2).ToList();
+            extraFiles = files2.Except(expectedFiles).ToList();
+
+            // test that output is what's expected
+            Assert.That(missingFiles.Count() == 0 && extraFiles.Count() == 0);
+
+            string[] checkResults = File.ReadAllLines(Path.Combine(thisTaskOutputFolder, "VariantOutput", "VariantAnalysisResultSummary.txt"));
+
+            Assert.AreEqual("Number of potentially variant containing peptides identified at 1% FDR: 1", checkResults[4]);
+            Assert.AreEqual("Number of unqiuely identified variant peptides at 1% FDR: 1", checkResults[5]);
+            Assert.AreEqual("Number of frameshift variant peptides at 1% FDR: 1", checkResults[7]);
+            Assert.AreEqual("Number of modified variant peptides at 1% FDR: 1", checkResults[13]);
```

Comment:
add checks for each of the kinds of variants

Reply:
done